src/shared/parse.cpp

```diff
@@ -4354,7 +4350,7 @@ bool Parse_ReadTokenHandle( int handle, pc_token_t *pc_token )
 Parse_SourceFileAndLine
 ===============
 */
-int Parse_SourceFileAndLine( int handle, char *filename, int *line )
+int Parse_SourceFileAndLine( int handle, char filename[ MAX_QPATH ], int *line )
```

Comment:
You could type the argument as `char(&filename)[MAX_QPATH]`, to enforce that it is actually an array of that length.

Reply:
that's a really neat trick. I was really surprised to see there was no way to get even a warning from the compiler with the sized array.